src/ec/curve25519/ops.rs

```diff
@@ -121,6 +121,72 @@ fn encode_point(x: Elem, y: Elem, z: Elem) -> EncodedPoint {
     bytes
 }
 
+#[cfg(all(target_arch = "x86_64", not(windows)))]
+pub fn scalar_mult_with_base_point(scalar: &Scalar) -> EncodedPoint {
+    let mut point = [0u8; ELEM_LEN];
+
+    unsafe { GFp_x25519_scalar_mult(&mut point, scalar, &BASE_POINT); }
+
+    point
+}
+
+#[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
```

Comment:
Hmm, yes. But, looking back at it, the ASM `GFp_x25519_NEON()` does not get compiled for AArch64, only for `__arm__`. This ASM function comes from the DJB implementation in SUPERCOP, which (in its `architectures` file) only lists `arm` and `armeabi`, with `aarch64` appearing in other implementations (so no equivocation is occurring).

So, this specialized implementation is only (possibly) appropriate for 32-bit ARM, and right now, we should actually use the generic base point implementation for AArch64, independent of the NEON support of that architecture.